src/audio/copier.c

```diff
@@ -252,53 +352,67 @@ static struct comp_dev *create_dai(struct comp_dev *parent_dev, struct copier_da
 		dai.type = SOF_DAI_INTEL_ALH;
 		dai.is_config_blob = true;
 		type = ipc4_gtw_alh;
-		/* copier id = (group id << 4) + codec id + IPC4_ALH_DAI_INDEX_OFFSET
-		 * dai_index = (group id << 8) + codec id;
+
+		/* copier
+		 * {
+		 *  gtw_cfg
+		 *  {
+		 *     gtw_node_id;
+		 *     config_length;
+		 *     config_data
+		 *     {
+		 *         count;
+		 *        {
+		 *           node_id;  \\ normal gtw id
+		 *           mask;
+		 *        }  mapping[MAX_ALH_COUNT];
+		 *     }
+		 *   }
+		 * }
 		 */
-		dai.dai_index = ((dai.dai_index & 0xF0) << DAI_NUM_ALH_BI_DIR_LINKS_GROUP) +
-			(dai.dai_index & 0xF) - IPC4_ALH_DAI_INDEX_OFFSET;
+		 /* get gtw node id in config data */
+		if (copier->gtw_cfg.config_length) {
+			alh_blob = (struct sof_alh_configuration_blob *)copier->gtw_cfg.config_data;
+			dai_count = alh_blob->alh_cfg.count;
+			for (i = 0; i < dai_count; i++)
+				dai_index[i] =
+				IPC4_ALH_DAI_INDEX(alh_blob->alh_cfg.mapping[i].alh_id);
+		} else {
+			dai_index[dai_count - 1] = IPC4_ALH_DAI_INDEX(dai_index[dai_count - 1]);
```

Comment:
`dai_index[dai_count - 1] = IPC4_ALH_DAI_INDEX(node_id.f.v_index)` is more readable.

Reply:
sure, thanks!